library/core/src/io/borrowed_buf.rs

```diff
@@ -341,10 +331,6 @@ impl<'a> BorrowedCursor<'a> {
             self.as_mut()[..buf.len()].write_copy_of_slice(buf);
         }
 
-        // SAFETY: We just added the entire contents of buf to the filled section.
-        unsafe {
-            self.set_init(buf.len());
```

Comment:
We could check if this append *fully* initialized the data, and mark it accordingly. Not sure if that's worth doing, but it seems straightforward.